server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/ScheduledTaskLoggerService.java

```diff
@@ -0,0 +1,224 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.oceanbase.odc.service.schedule;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.util.Optional;
+
+import org.apache.commons.io.FileUtils;
+import org.springframework.stereotype.Service;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.oceanbase.odc.common.json.JsonUtils;
+import com.oceanbase.odc.core.shared.PreConditions;
+import com.oceanbase.odc.core.shared.exception.UnexpectedException;
+import com.oceanbase.odc.metadb.schedule.ScheduleTaskEntity;
+import com.oceanbase.odc.metadb.task.JobEntity;
+import com.oceanbase.odc.service.common.response.SuccessResponse;
+import com.oceanbase.odc.service.dispatch.DispatchResponse;
+import com.oceanbase.odc.service.dispatch.JobDispatchChecker;
+import com.oceanbase.odc.service.dispatch.RequestDispatcher;
+import com.oceanbase.odc.service.dispatch.TaskDispatchChecker;
+import com.oceanbase.odc.service.objectstorage.cloud.CloudObjectStorageService;
+import com.oceanbase.odc.service.task.caller.ExecutorIdentifier;
+import com.oceanbase.odc.service.task.caller.ExecutorIdentifierParser;
+import com.oceanbase.odc.service.task.config.TaskFrameworkEnabledProperties;
+import com.oceanbase.odc.service.task.constants.JobAttributeKeyConstants;
+import com.oceanbase.odc.service.task.executor.logger.LogUtils;
+import com.oceanbase.odc.service.task.model.ExecutorInfo;
+import com.oceanbase.odc.service.task.model.OdcTaskLogLevel;
+import com.oceanbase.odc.service.task.service.TaskFrameworkService;
+import com.oceanbase.odc.service.task.util.JobUtils;
+import com.oceanbase.odc.service.task.util.TaskExecutorClient;
+
+import cn.hutool.core.io.FileUtil;
+import cn.hutool.core.util.ObjectUtil;
+import cn.hutool.core.util.StrUtil;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * @author mayang
+ */
+@Service
+@Slf4j
+public class ScheduledTaskLoggerService {
+
+    private static final String LOG_PATH_PATTERN = "%s/scheduleTask/%s-%s/%s/log.%s";
+
+    private final ScheduleTaskService scheduleTaskService;
+    private final TaskFrameworkEnabledProperties taskFrameworkEnabledProperties;
+    private final RequestDispatcher requestDispatcher;
+    private final TaskDispatchChecker dispatchChecker;
+    private final TaskFrameworkService taskFrameworkService;
+    private final JobDispatchChecker jobDispatchChecker;
+    private final TaskExecutorClient taskExecutorClient;
+
+    private final LoggerProperty loggerProperty;
+    private final CloudObjectStorageService cloudObjectStorageService;
+
+    public ScheduledTaskLoggerService(ScheduleTaskService scheduleTaskService,
```

Comment:
what is the reason use constructor here ?

Reply:
using constructor injection gives more flexibility in managing dependencies and avoids adding too many @Autowired annotations to your classes，it is functionally equivalent to @Autowired